M2/Macaulay2/packages/Hadamard.m2

```diff
@@ -0,0 +1,441 @@
+newPackage(
+    "Hadamard",
+    Version => "1.0",
+    Date => "November 2020",
+    Authors => {
+
+	{Name => "Iman Bahmani Jafarloo",
+	    Email => "ibahmani89@gmail.com",
+	    HomePage => "http://calvino.polito.it/~imanbj"
+	    }
+	}, -- TODO
+    Headline => "A package for the Hadamard products of varieties",
+    AuxiliaryFiles => false,
+    DebuggingMode => false,
+    Reload => false,
+    PackageExports => {"Points"},
+    Keywords => {"Commutative Algebra"}
+    )
+export {
+    -- types
+    "Point",
+    -- methods
+    "point",
+    "hadamardProduct",
+    "hadamardPower",
+    "idealOfProjectivePoints"
+    }
+
+--defining a new type of objects: points
+
+Point = new Type of BasicList
+
+point=method()
+point(VisibleList):=(P)->(
+    if all(P, i->i==0) then return "error: all entries are zero" else
+    if #delete(null, P) =!= #P then return "error: nuul entries are not allowed"
+    else new Point from P)
+
+makeListOfPoints=method()
+makeListOfPoints(VisibleList):=(L)->(
+             apply(L,P-> point P)
+	     )
+
+Point * Point:=(p,q)->(
+    if #p =!= #q then error("the points should be in a same projective space") else
+    pp:=apply(p,q,times);
+    if all(pp, i->i==0) then error("this operation is not well-defined in projective space")
```

Comment:
The error message refers to the operation instead of its result.  Better to say something like `error "product of points has no nonzero coordinate"`.